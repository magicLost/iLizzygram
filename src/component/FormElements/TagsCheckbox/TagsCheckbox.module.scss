.root {
  .formControl {
    padding-top: 10px;
    .formGroup {
      display: flex;
      flex-wrap: wrap;
      flex-direction: row;
      flex-grow: 0;
      flex-shrink: 0;

      .container {
        display: flex;
        margin-right: 5px;

        .skeleton {
          margin-right: 10px;
        }
      }
    }
  }
}
