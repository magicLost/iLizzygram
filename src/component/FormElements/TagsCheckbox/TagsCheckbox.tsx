import React, { useEffect, useState } from "react";
import classes from "./TagsCheckbox.module.scss";
//import { createStyles, makeStyles, Theme } from "@material-ui/core/styles";
import FormLabel from "@material-ui/core/FormLabel";
import Skeleton from "@material-ui/lab/Skeleton";
import FormControl from "@material-ui/core/FormControl";
import FormGroup from "@material-ui/core/FormGroup";
import FormControlLabel from "@material-ui/core/FormControlLabel";
import FormHelperText from "@material-ui/core/FormHelperText";
import Checkbox from "@material-ui/core/Checkbox";
import { useQuery, ApolloError } from "@apollo/client";
import gql from "graphql-tag";
//import { ApolloError } from "";

export interface ITag {
  _id: string;
  title: string;
  name: string;
}

interface ITagsCheckboxProps {
  register: any;
  rules: any;
  setValue: (name: string, data: any) => void;
  watch: (stateName: string) => { [tag_id: string]: boolean };
  clearError: (name: string) => void;
  disabled: boolean;
  error: any;
  label: string;
  defaultTagsIds?: string[];
}

export const TAGS = gql`
  {
    tags {
      _id
      name
      title
    }
  }
`;

export const getCheckboxes = (
  data: any,
  handleChange: (event: any) => void,
  state: { [name: string]: boolean },
  loading: boolean,
  queryError: ApolloError,
  disabled: boolean
) => {
  console.log("[getCheckboxes]");
  if (
    loading ||
    queryError ||
    !state ||
    state[data.tags[0]._id] === undefined
  ) {
    return [1, 2, 3, 4, 5, 6, 7].map(value => {
      return (
        <div key={classes.container + value} className={classes.container}>
          <Skeleton className={classes.skeleton} width={20} height={30} />
          <Skeleton className={classes.skeleton} width={65} height={30} />
        </div>
      );
    });
  }

  //console.log("[getCheckboxes] RENDER DATA");
  if (data && data.tags) {
    return data.tags.map((tag, index) => {
      return (
        <FormControlLabel
          key={tag._id + index}
          control={
            <Checkbox
              checked={state[tag._id]}
              onChange={handleChange}
              name={tag._id}
              disabled={disabled}
            />
          }
          label={tag.title}
        />
      );
    });
  }
};

const TagsCheckbox = ({
  register,
  setValue,
  watch,
  clearError,
  disabled,
  error,
  rules,
  label,
  defaultTagsIds,
}: ITagsCheckboxProps) => {
  //const classes = useStyles();

  useEffect(() => {
    register({ name: "tags", type: "custom" }, rules);
  }, [register]);

  // tagsState = { tag_id: boolean } - in input checkbox we use name={tag._id}
  const tagsState = watch("tags");

  const { data, loading, error: queryError } = useQuery(TAGS, {
    onCompleted: data => {
      const initState: { [name: string]: boolean } = {};
      const length = defaultTagsIds ? defaultTagsIds.length : 0;
      let tagsMarked = 0;
      data.tags.forEach(tag => {
        if (length > 0 && tagsMarked < length) {
          /*  console.log(
            "TAGS DEFAULT STATE CALC",
            length,
            tagsMarked,
            defaultTagsIds.includes(tag._id)
          ); */
          if (defaultTagsIds.includes(tag._id)) {
            initState[tag._id] = true;
            tagsMarked++;
          } else {
            initState[tag._id] = false;
          }
        } else {
          initState[tag._id] = false;
        }
      });
      //console.log("TAGS DEFAULT STATE", initState);
      setValue("tags", initState);
      //setState(initState);
    },
  });

  const handleChange = (event: any) => {
    //console.log("handleDateChange", event.target);
    //const newState = { ...state, [event.target.name]: event.target.checked };
    const newState = {
      ...tagsState,
      [event.target.name]: event.target.checked,
    };
    clearError("tags");
    setValue("tags", newState);
    //setState(newState);
  };

  const checkboxes = getCheckboxes(
    data,
    handleChange,
    //state,
    tagsState,
    loading,
    queryError,
    disabled
  );

  //loading, data, queryError, tagsState
  console.log("[RENDER] TagsCheckbox ");

  return (
    <div className={classes.root}>
      <FormControl
        fullWidth
        variant="filled"
        component="fieldset"
        error={error ? true : false}
        className={classes.formControl}
      >
        <FormLabel component="legend">{label}</FormLabel>
        <FormGroup className={classes.formGroup}>{checkboxes}</FormGroup>
        <FormHelperText>{error && error.message}</FormHelperText>
      </FormControl>
      {/* 
      <button onClick={() => console.log(watch("tags"))}>See</button> */}
    </div>
  );
};

export default TagsCheckbox;
