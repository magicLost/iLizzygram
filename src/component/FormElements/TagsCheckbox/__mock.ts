/* export const tagsData = [
  { _id: "123wsdf347423", title: "на улице", name: "street" },
  { _id: "123wsdf343423", title: "улыбка", name: "smile" },
  { _id: "123wsdd343423", title: "дача", name: "dacha" },
  { _id: "123wsdfj43423", title: "на природе", name: "nature" },
  { _id: "123wsdf34df23", title: "дома", name: "home" },
  { _id: "12wwsdf343423", title: "с петами", name: "pets" },
]; */

/* export const tagsData = new Map([
  ["123wsdf347423", { title: "на улице", name: "street" }],
  ["123wsdf343423", { title: "улыбка", name: "smile" }],
  ["123wsdd343423", { title: "дача", name: "dacha" }],
  ["123wsdfj43423", { title: "на природе", name: "nature" }],
  ["123wsdf34df23", { title: "дома", name: "home" }],
  ["12wwsdf343423", { title: "с петами", name: "pets" }],
]); */

export const tagsData = new Map([
  ["Hrj1grEKx6oM9Z1ZGP0G", { title: "зюганов", name: "zuganov" }],
  ["L45RiBaK18AEoyVekFQT", { name: "pets", title: "с животными" }],
  ["Pa8GvtwrT1tMDgNLwy4S", { title: "на улице", name: "street" }],
  ["Ql2r2DFzzjZnzP2adh9Z", { name: "smile", title: "улыбка" }],
  ["YBa0wyeWwEB6takyExmF", { title: "задумчиво", name: "thoughtfully" }],
  ["YxX09wTx6kWOfZQ0ORFs", { title: "дома", name: "home" }],
  ["cdbI7sOCFVFv337chtBE", { title: "на природе", name: "nature" }],
]);

export const state = {
  Hrj1grEKx6oM9Z1ZGP0G: false,
  L45RiBaK18AEoyVekFQT: false,
  Pa8GvtwrT1tMDgNLwy4S: false,
  Ql2r2DFzzjZnzP2adh9Z: false,
  YBa0wyeWwEB6takyExmF: false,
  YxX09wTx6kWOfZQ0ORFs: false,
  cdbI7sOCFVFv337chtBE: false,
};

export const defaultTagsIds = [
  "Pa8GvtwrT1tMDgNLwy4S",
  "YBa0wyeWwEB6takyExmF",
  "L45RiBaK18AEoyVekFQT",
];
