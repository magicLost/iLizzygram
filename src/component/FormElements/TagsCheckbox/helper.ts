/* import { useEffect } from "react";
import { ICheckboxItemData } from ".";
import { TTagsData } from "./../../../store/types";

/* export interface ITag extends ICheckboxItemData {}

export interface ITagsResponseData {
  tags: ITag[];
} 

export type TAGS_STATE = { [id: string]: boolean };

export const useTags = (
  setInitState: (initState: TAGS_STATE) => void | undefined,
  tagsData?: TTagsData,
  defaultTagsIds?: string[]
) => {
  useEffect(() => {
    if (tagsData) {
      const initState: TAGS_STATE = {};
      const length = defaultTagsIds ? defaultTagsIds.length : 0;
      let tagsMarked = 0;
      tagsData.forEach((tagData, id) => {
        if (length > 0 && tagsMarked < length) {
          if (defaultTagsIds.includes(id)) {
            initState[id] = true;
            tagsMarked++;
          } else {
            initState[id] = false;
          }
        } else {
          initState[id] = false;
        }
      });
      //console.log("TAGS DEFAULT STATE", initState);
      setInitState(initState);
    }
  }, [tagsData]);
};

export const initTagsState = (
  setInitState: (initState: TAGS_STATE) => void | undefined,
  tagsData?: TTagsData,
  defaultTagsIds?: string[]
) => {
  if (tagsData) {
    const initState: TAGS_STATE = {};
    const length = defaultTagsIds ? defaultTagsIds.length : 0;
    let tagsMarked = 0;
    tagsData.forEach((tagData, id) => {
      if (length > 0 && tagsMarked < length) {
        if (defaultTagsIds.includes(id)) {
          initState[id] = true;
          tagsMarked++;
        } else {
          initState[id] = false;
        }
      } else {
        initState[id] = false;
      }
    });
    //console.log("TAGS DEFAULT STATE", initState);
    setInitState(initState);
  }
};
 */
