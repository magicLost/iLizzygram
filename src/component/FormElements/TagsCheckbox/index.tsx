import React, { FC } from "react";
import TagsCheckboxWidget, { ITagsCheckboxProps } from "./TagsCheckbox";
import { useTags } from "../../../store/hooks";

//import gql from "graphql-tag";
//import { ApolloError } from "";

export interface ICheckboxItemData {
  //_id: string;
  title: string;
  name: string;
}

export const TagsCheckbox: FC<ITagsCheckboxProps> = props => {
  const { tagsState } = useTags();

  //loading, data, queryError, tagsState
  console.log("[RENDER TAGS CHECKBOX] ");

  return <TagsCheckboxWidget tagsState={tagsState} {...props} />;
};

export default TagsCheckbox;
