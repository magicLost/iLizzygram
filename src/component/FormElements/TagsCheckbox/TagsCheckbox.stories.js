import React from "react";
import { action } from "@storybook/addon-actions";
import TagsCheckbox, { getCheckboxes, TAGS } from "./TagsCheckbox";
import classes from "./TagsCheckbox.module.scss";
import FormControl from "@material-ui/core/FormControl";
import FormGroup from "@material-ui/core/FormGroup";
import FormLabel from "@material-ui/core/FormLabel";
import FormHelperText from "@material-ui/core/FormHelperText";
import { useForm } from "react-hook-form";
import { tagsRules } from "../../../container/forms/rules";
import { MockedProvider } from "@apollo/client/testing";

export default {
  component: TagsCheckbox,
  title: "FormElements/TagsCheckbox",
  decorators: [
    (story) => (
      <div style={{ width: "500px", margin: "auto", paddingTop: "20px" }}>
        {story()}
      </div>
    ),
  ],
  //decorators: [withKnobs],
  // Our exports that end in "Data" are not stories.
  excludeStories: /.*Data$/,
};

export const tagsData = [
  { _id: "123wsdf347423", title: "на улице", name: "street" },
  { _id: "123wsdf343423", title: "улыбка", name: "smile" },
  { _id: "123wsdd343423", title: "дача", name: "dacha" },
  { _id: "123wsdfj43423", title: "на природе", name: "nature" },
  { _id: "123wsdf34df23", title: "дома", name: "home" },
  { _id: "12wwsdf343423", title: "с петами", name: "pets" },
];

export const mockQueriesData = [
  {
    request: {
      query: TAGS,
    },
    result: {
      data: {
        tags: tagsData,
      },
    },
  },
];

const state = {
  "123wsdf347423": false,
  "123wsdf343423": false,
  "123wsdd343423": false,
  "123wsdfj43423": false,
  "123wsdf34df23": false,
  "12wwsdf343423": false,
};

export const Default = () => {
  const {
    register,
    handleSubmit,
    setValue,
    clearError,
    watch,
    errors,
  } = useForm();

  return (
    <MockedProvider mocks={mockQueriesData} addTypename={true}>
      <form onSubmit={handleSubmit((data) => console.log(data))}>
        <TagsCheckbox
          label={"Choose fucking tags:"}
          register={register}
          rules={tagsRules}
          setValue={setValue}
          watch={watch}
          clearError={clearError}
          error={errors.tags}
          disabled={false}
          defaultTagsIds={["123wsdfj43423", "12wwsdf343423"]}
        />
        <br />
        <button type="submit">Go</button>
      </form>
    </MockedProvider>
  );
};

export const LoadingTags = () => {
  const checkboxes = getCheckboxes(
    { tags: tagsData },
    () => {},
    state,
    true,
    null,
    false
  );

  return (
    <div className={classes.root}>
      <FormControl
        fullWidth
        variant="filled"
        component="fieldset"
        error={false}
        className={classes.formControl}
      >
        <FormLabel component="legend">{"Добавьте тэги к фоте:"}</FormLabel>
        <FormGroup className={classes.formGroup}>{checkboxes}</FormGroup>
        <FormHelperText>{""}</FormHelperText>
      </FormControl>
    </div>
  );
};

export const Tags = () => {
  const checkboxes = getCheckboxes(
    { tags: tagsData },
    () => {},
    state,
    false,
    null,
    false
  );

  return (
    <div className={classes.root}>
      <FormControl
        fullWidth
        variant="filled"
        component="fieldset"
        error={false}
        className={classes.formControl}
      >
        <FormLabel component="legend">{"Добавьте тэги к фоте:"}</FormLabel>
        <FormGroup className={classes.formGroup}>{checkboxes}</FormGroup>
        <FormHelperText>{""}</FormHelperText>
      </FormControl>
    </div>
  );
};
