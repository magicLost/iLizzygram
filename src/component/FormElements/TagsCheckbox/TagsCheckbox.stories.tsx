import React, { useEffect } from "react";
import { action } from "@storybook/addon-actions";
import { TagsCheckbox } from "./TagsCheckbox";
/* import classes from "./TagsCheckbox.module.scss";
import FormControl from "@material-ui/core/FormControl";
import FormGroup from "@material-ui/core/FormGroup";
import FormLabel from "@material-ui/core/FormLabel";
import FormHelperText from "@material-ui/core/FormHelperText"; */
import { useForm } from "react-hook-form";
import { tagsRules } from "../../../photos/form/Photo.rules";
//import { MockedProvider } from "@apollo/client/testing";
//import { useQuery, ApolloError } from "@apollo/client";
//import { useTags } from "../../../hooks/photos/useTags";
import { tagsData, state as initTagsState, defaultTagsIds } from "./__mock";

export default {
  component: TagsCheckbox,
  title: "FormElements/TagsCheckbox",
  decorators: [
    story => (
      <div style={{ width: "500px", margin: "auto", paddingTop: "20px" }}>
        {story()}
      </div>
    ),
  ],
  //decorators: [withKnobs],
  // Our exports that end in "Data" are not stories.
  excludeStories: /.*Data$/,
};

const Form = () => {
  const {
    register,
    handleSubmit,
    setValue,
    clearErrors,
    watch,
    errors,
  } = useForm({
    defaultValues: {
      tags: initTagsState,
    },
  });

  useEffect(() => {
    register({ name: "tags", type: "custom" }, tagsRules);
  }, [register]);

  // tagsState = { tag_id: boolean } - in input checkbox we use name={tag._id}
  const tagsState = watch("tags");

  const onCheckboxChange = event => {
    //console.log("handleDateChange", event.target);
    //const newState = { ...state, [event.target.name]: event.target.checked };
    const newState = {
      ...tagsState,
      [event.target.name]: event.target.checked,
    };
    clearErrors("tags");
    setValue("tags", newState);
    //setState(newState);
  };

  console.log("RENDER TAGS STORIES FORM");

  return (
    <form onSubmit={handleSubmit(data => console.log("SUBMIT", data))}>
      <TagsCheckbox
        label={"Choose tags:"}
        tagsState={{
          tags: tagsData,
          error: false,
          loading: false,
        }}
        itemsState={tagsState}
        onChange={onCheckboxChange}
        error={errors.tags}
        disabled={false}
      />
      <br />
      <button type="submit">Go</button>
    </form>
  );
};

export const Default = () => {
  return <Form />;
};

const Template = args => <TagsCheckbox {...args} />;

const args = {
  label: "Choose tags:",
  tagsState: {
    tags: tagsData,
    error: false,
    loading: false,
  },
  itemsState: initTagsState,
  onChange: () => console.log("onChange"),
  error: undefined,
  disabled: false,
};

export const Tags = Template.bind({});
Tags.args = {
  ...args,
};

export const ErrorTags = Template.bind({});
ErrorTags.args = {
  ...args,
  error: { message: "Вы что с ума сошли?" },
};

export const LoadingTags = Template.bind({});
LoadingTags.args = {
  ...args,
  tagsState: {
    ...args.tagsState,
    loading: true,
  },
};

/* export const LoadingTags = () => {
  const checkboxes = getCheckboxes(
    () => {},
    initTagsState,
    true,
    null,
    false,
    tagsData
  );

  return (
    <div className={classes.root}>
      <FormControl
        fullWidth
        variant="filled"
        component="fieldset"
        error={false}
        className={classes.formControl}
      >
        <FormLabel component="legend">{"Добавьте тэги к фоте:"}</FormLabel>
        <FormGroup className={classes.formGroup}>{checkboxes}</FormGroup>
        <FormHelperText>{""}</FormHelperText>
      </FormControl>
    </div>
  );
}; */

/* export const Tags = () => {
  const checkboxes = getCheckboxes(
    () => {},
    initTagsState,
    false,
    null,
    false,
    tagsData
  );

  return (
    <div className={classes.root}>
      <FormControl
        fullWidth
        variant="filled"
        component="fieldset"
        error={false}
        className={classes.formControl}
      >
        <FormLabel component="legend">{"Добавьте тэги к фоте:"}</FormLabel>
        <FormGroup className={classes.formGroup}>{checkboxes}</FormGroup>
        <FormHelperText>{""}</FormHelperText>
      </FormControl>
    </div>
  );
}; */
