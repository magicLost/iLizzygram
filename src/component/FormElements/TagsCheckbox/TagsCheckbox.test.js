import React, { useState } from "react";
import { render, fireEvent, cleanup } from "@testing-library/react";
import { configure } from "@testing-library/dom";
import "@testing-library/jest-dom/extend-expect";
import { MockedProvider } from "@apollo/client/testing";
import TagsCheckbox, { TAGS } from ".";
import classes from "./TagsCheckbox.module.scss";
import { act } from "react-dom/test-utils";
import wait from "waait";
import {
  defaultTagsIds,
  tagsData,
  state as tagsState,
} from "../../../hooks/photos/useTags.mock";

const label = "My tags";

const onCheckboxChange = jest.fn();
//const setValue = jest.fn();
const register = jest.fn();
//const watch = jest.fn(() => tagsState);
const tagsRules = {};

//const ddefaultTagsIds = ["123wsdf343423", "123wsdf34df23", "12wwsdf343423"];

/* const Wrapper = ({ defaultTagsIds }) => {
  const [tagsState, setTagsState] = useState({
    /* "123wsdf347423": false,
    "123wsdf343423": false,
    "123wsdd343423": false,
    "123wsdfj43423": false,
    "123wsdf34df23": false,
    "12wwsdf343423": false, 
  });

  const setValue = (name, state) => {
    //console.log("setTagsState", state);
    setTagsState(state);
  };

  const watch = name => {
    return tagsState;
  };

  return (
    <TagsCheckbox
      items={data ? data.tags : data}
      itemsState={tagsState}
      loading={loading}
      queryError={false}
      onChange={onCheckboxChange}
      error={undefined}
      disabled={false}
      label={label}
      defaultTagsIds={defaultTagsIds}
    />
  );
}; */

describe("TagsCheckbox", () => {
  let _render = null;

  describe("Render and props test", () => {
    beforeEach(() => {});

    afterEach(cleanup);

    describe("", () => {
      test("It must render label and Sceletons on loading checkboxes", async () => {
        _render = render(
          <TagsCheckbox
            items={tagsData}
            itemsState={tagsState}
            loading={true}
            queryError={false}
            onChange={onCheckboxChange}
            error={undefined}
            disabled={false}
            label={label}
          />
        );

        const tagsCheckbox = _render.getAllByText(label);

        let tagsFormGroup = document.querySelector(
          `div[class="MuiFormGroup-root ${classes.formGroup}"]`
        );

        expect(tagsCheckbox).toHaveLength(1);

        expect(tagsFormGroup.children).toHaveLength(7);
        expect(tagsFormGroup.children[0].className).toEqual("container");
      });

      test("It must set checked to true on defaultTagsIds", async () => {
        const state = { ...tagsState };
        state["123wsdd343423"] = true;
        state["123wsdf34df23"] = true;
        _render = render(
          <TagsCheckbox
            items={tagsData}
            itemsState={state}
            loading={false}
            queryError={false}
            onChange={onCheckboxChange}
            error={undefined}
            disabled={false}
            label={label}
          />
        );

        let tagsFormGroup = document.querySelector(
          `div[class="MuiFormGroup-root ${classes.formGroup}"]`
        );

        expect(tagsFormGroup.children).toHaveLength(6);
        expect(tagsFormGroup.children[0].className).toEqual(
          "MuiFormControlLabel-root"
        );

        expect(
          tagsFormGroup.children[1].querySelector("input").checked
        ).toEqual(false);
        expect(
          tagsFormGroup.children[0].querySelector("input").checked
        ).toEqual(false);
        expect(
          tagsFormGroup.children[2].querySelector("input").checked
        ).toEqual(true);
        expect(
          tagsFormGroup.children[3].querySelector("input").checked
        ).toEqual(false);
        expect(
          tagsFormGroup.children[4].querySelector("input").checked
        ).toEqual(true);
        expect(
          tagsFormGroup.children[5].querySelector("input").checked
        ).toEqual(false);
      });
    });

    describe("Snapshots", () => {
      test("We got skeletons on load snapshot", () => {
        _render = render(
          <TagsCheckbox
            items={tagsData}
            itemsState={tagsState}
            loading={true}
            queryError={false}
            onChange={onCheckboxChange}
            error={undefined}
            disabled={false}
            label={label}
          />
        );

        const { baseElement } = _render;
        expect(baseElement).toMatchSnapshot();
      });

      test("We got checkboxes on load data snapshot", async () => {
        const state = { ...tagsState };
        state["123wsdd343423"] = true;
        state["123wsdf34df23"] = true;
        _render = render(
          <TagsCheckbox
            items={tagsData}
            itemsState={state}
            loading={false}
            queryError={false}
            onChange={onCheckboxChange}
            error={undefined}
            disabled={false}
            label={label}
          />
        );

        const { baseElement } = _render;
        expect(baseElement).toMatchSnapshot();
      });
      /*   test("We got checkboxes with default values on load data snapshot", async () => {
        _render.rerender(
          <MockedProvider mocks={mockQueries} addTypename={false}>
            <Wrapper defaultTagsIds={ddefaultTagsIds} />
          </MockedProvider>
        );

        await act(async () => {
          await wait(0);

          await wait(0);

          const { baseElement } = _render;
          expect(baseElement).toMatchSnapshot();
        });
      }); */
    });
  });
});
