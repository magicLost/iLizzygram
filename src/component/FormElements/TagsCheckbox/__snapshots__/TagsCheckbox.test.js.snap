// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagsCheckbox Render and props test Snapshots We got checkboxes on load data snapshot 1`] = `
<body>
  <div>
    <div
      class="root"
    >
      <fieldset
        class="MuiFormControl-root formControl MuiFormControl-fullWidth"
      >
        <legend
          class="MuiFormLabel-root"
        >
          My tags
        </legend>
        <div
          class="MuiFormGroup-root formGroup"
        >
          <label
            class="MuiFormControlLabel-root"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-5 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  class="PrivateSwitchBase-input-8"
                  data-indeterminate="false"
                  name="123wsdf347423"
                  type="checkbox"
                  value=""
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              на улице
            </span>
          </label>
          <label
            class="MuiFormControlLabel-root"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-5 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  class="PrivateSwitchBase-input-8"
                  data-indeterminate="false"
                  name="123wsdf343423"
                  type="checkbox"
                  value=""
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              улыбка
            </span>
          </label>
          <label
            class="MuiFormControlLabel-root"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-5 MuiCheckbox-root MuiCheckbox-colorSecondary PrivateSwitchBase-checked-6 Mui-checked MuiIconButton-colorSecondary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  checked=""
                  class="PrivateSwitchBase-input-8"
                  data-indeterminate="false"
                  name="123wsdd343423"
                  type="checkbox"
                  value=""
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              дача
            </span>
          </label>
          <label
            class="MuiFormControlLabel-root"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-5 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  class="PrivateSwitchBase-input-8"
                  data-indeterminate="false"
                  name="123wsdfj43423"
                  type="checkbox"
                  value=""
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              на природе
            </span>
          </label>
          <label
            class="MuiFormControlLabel-root"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-5 MuiCheckbox-root MuiCheckbox-colorSecondary PrivateSwitchBase-checked-6 Mui-checked MuiIconButton-colorSecondary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  checked=""
                  class="PrivateSwitchBase-input-8"
                  data-indeterminate="false"
                  name="123wsdf34df23"
                  type="checkbox"
                  value=""
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              дома
            </span>
          </label>
          <label
            class="MuiFormControlLabel-root"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-5 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  class="PrivateSwitchBase-input-8"
                  data-indeterminate="false"
                  name="12wwsdf343423"
                  type="checkbox"
                  value=""
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              с петами
            </span>
          </label>
        </div>
        <p
          class="MuiFormHelperText-root MuiFormHelperText-contained"
        />
      </fieldset>
    </div>
  </div>
</body>
`;

exports[`TagsCheckbox Render and props test Snapshots We got skeletons on load snapshot 1`] = `
<body>
  <div>
    <div
      class="root"
    >
      <fieldset
        class="MuiFormControl-root formControl MuiFormControl-fullWidth"
      >
        <legend
          class="MuiFormLabel-root"
        >
          My tags
        </legend>
        <div
          class="MuiFormGroup-root formGroup"
        >
          <div
            class="container"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text skeleton MuiSkeleton-pulse"
              style="width: 20px; height: 30px;"
            />
            <span
              class="MuiSkeleton-root MuiSkeleton-text skeleton MuiSkeleton-pulse"
              style="width: 65px; height: 30px;"
            />
          </div>
          <div
            class="container"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text skeleton MuiSkeleton-pulse"
              style="width: 20px; height: 30px;"
            />
            <span
              class="MuiSkeleton-root MuiSkeleton-text skeleton MuiSkeleton-pulse"
              style="width: 65px; height: 30px;"
            />
          </div>
          <div
            class="container"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text skeleton MuiSkeleton-pulse"
              style="width: 20px; height: 30px;"
            />
            <span
              class="MuiSkeleton-root MuiSkeleton-text skeleton MuiSkeleton-pulse"
              style="width: 65px; height: 30px;"
            />
          </div>
          <div
            class="container"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text skeleton MuiSkeleton-pulse"
              style="width: 20px; height: 30px;"
            />
            <span
              class="MuiSkeleton-root MuiSkeleton-text skeleton MuiSkeleton-pulse"
              style="width: 65px; height: 30px;"
            />
          </div>
          <div
            class="container"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text skeleton MuiSkeleton-pulse"
              style="width: 20px; height: 30px;"
            />
            <span
              class="MuiSkeleton-root MuiSkeleton-text skeleton MuiSkeleton-pulse"
              style="width: 65px; height: 30px;"
            />
          </div>
          <div
            class="container"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text skeleton MuiSkeleton-pulse"
              style="width: 20px; height: 30px;"
            />
            <span
              class="MuiSkeleton-root MuiSkeleton-text skeleton MuiSkeleton-pulse"
              style="width: 65px; height: 30px;"
            />
          </div>
          <div
            class="container"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text skeleton MuiSkeleton-pulse"
              style="width: 20px; height: 30px;"
            />
            <span
              class="MuiSkeleton-root MuiSkeleton-text skeleton MuiSkeleton-pulse"
              style="width: 65px; height: 30px;"
            />
          </div>
        </div>
        <p
          class="MuiFormHelperText-root MuiFormHelperText-contained"
        />
      </fieldset>
    </div>
  </div>
</body>
`;
